### ISSUE TITLE
Ubuntu 18.04 RDP sesseion ends Whatever xorg xvnc

### ISSUE THREAD

**Issue by TacXingXing** [neutrinolabs/xrdp#1169]
Log:
```
20180718-19:22:15] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:22:58] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:25:07] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:25:41] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-19:25:41] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-19:25:45] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-19:25:45] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-19:25:49] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-19:25:49] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-19:25:53] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-19:25:53] [ERROR] xrdp_mm_connect_chansrv: error in trans_connect chan
[20180718-19:25:54] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-19:25:54] [ERROR] xrdp_mm_connect_chansrv: error in trans_connect chan
[20180718-19:29:18] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:29:37] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:34:46] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:35:01] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:35:26] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:43:23] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:51:46] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:51:52] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:52:04] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-19:59:32] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-20:11:29] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-20:12:57] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-20:14:42] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-20:15:04] [ERROR] xrdp_wm_log_msg: Error connecting to: 10.114.1.109
[20180718-20:15:08] [ERROR] xrdp_wm_log_msg: Error connecting to: 10.114.1.109
[20180718-20:15:23] [ERROR] xrdp_wm_log_msg: Error connecting to: 10.114.1.109
[20180718-20:15:38] [ERROR] xrdp_wm_log_msg: Error connecting to: 127.0.0.1
[20180718-20:35:58] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-20:36:15] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-20:36:19] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-20:36:24] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-20:36:28] [ERROR] xrdp_mm_connect_chansrv: connect failed trying again...
[20180718-20:36:28] [ERROR] xrdp_mm_connect_chansrv: error in trans_connect chan
[20180718-20:48:19] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-20:54:23] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20180718-20:59:15] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
```

**Comment by metalefty** [comment]
Can't you really understand what the log says?

**Comment by TacXingXing** [comment]
Yes，But....I don't know how to fix it. I find anothers have the same problem like me .

**Comment by metalefty** [comment]
Really, just adjust permissions of the file. It's very common knowledge, not an xrdp issue.

**Comment by compdoc** [comment]
What a horrible response. Be nice to know how to fix this, instead of flip replies.

**Comment by metalefty** [comment]
man chmod

**Comment by kovalexpro** [comment]
ls -la /etc/xrdp/key.pem
lrwxrwxrwx 1 root root 38 Oct 20 02:57 /etc/xrdp/key.pem -> /etc/ssl/private/ssl-cert-snakeoil.key

so what are we supposed to adjust there? chown?!

**Comment by x-strong** [comment]
@kovalexpro May be you should try to change the exactly file /etc/ssl/private/ssl-cert-snakeoil.key mode.
I met the same problem and solved it by doing that. 
But I still  can't login in the enhanced mode by Xorg : (

**Comment by raupie** [comment]
I had the same warning in my xrdp.log.

My `/etc/ssl/private` folder was set at 710 and I changed it to 755.

The ssl-cert-snakeoil.key and /etc/xrdp/key.pem symlink was 777.

**Comment by metalefty** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=856436#15 

> For now, I think the local administrator should add xrdp to the ssl-cert
> group if they want to use TLS. This is IMHO not a bug in the package,
> because by default, xrdp also uses RDP security and **adding daemon users**
> **to ssl-cert is a common and well-known practice**.

**Comment by sql-sith** [comment]
> **adding daemon users to ssl-cert is a common and well-known practice**.
You are correct, but if the daemon needs the permissions of a group, the installation of the daemon should take care of that. It should not be something that every user has to do as a post-install fix. Even your own advice has changed here from adjusting file permissions to adjusting group memberships. If you are somewhat unsure of the best route to follow, I don't think we should expect people with less experience to automatically know what to do.

In short, whatever the preferred method of making this work is, this is clearly a dependency of the xrdp service, so xrdp should install in a way that fulfills that dependency.

That's my 0.02. YMMV.

**Comment by metalefty** [comment]
> so xrdp should install in a way that fulfills that dependency

Ubuntu should do that. Not we xrdp team. The xrdp package is not created by us.
Please understand we xrdp team are not responsible for each distribution's packages. 

@sql-sith You should talk to Ubuntu or Debian team.

**Comment by metalefty** [comment]
ALL DISTRO PACKAGE SPECIFIC ISSUE SHOULD BE QUERIED DISTRO TEAM.

**Comment by metalefty** [comment]
I understand not a few people are in trouble because of SSL cert permission issue but that is really really not our issue at all. It is distro specific issue. Please please don't ask/complain us.  Ask the right people.
